components/Step4/Step4.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {useEffect, useState} from 'react';
+import {Pressable, ScrollView, Text, View} from 'react-native';
+import {Input} from '../../components';
+import {selectWizard, updateUserData} from '../../redux/reducers/WizardData';
+import {useDispatch, useSelector} from 'react-redux';
+import {fitnessWorkouts} from '../../utils/fitnessConstants';
+import {FontAwesomeIcon} from '@fortawesome/react-native-fontawesome';
+import {faCheck} from '@fortawesome/free-solid-svg-icons';
+import {Colors} from '../../utils/colors';
+import {wizard} from '../../utils/constants';
+
+import globalStyle from '../../assets/styles/globalStyle';
+import style from './style';
+
+const Step4 = () => {
+  const dispatch = useDispatch();
+  const [customWorkoutValue, setCustomWorkoutValue] = useState('');
+  const wizardData = useSelector(selectWizard);
+  const [fitnessCategoriesWorkout, setFitnessCategoriesWorkout] = useState<
+    Array<{label: string; checked: boolean}>
+  >([...fitnessWorkouts]);
+
+  useEffect(() => {
+    let unselectedItems: Array<{label: string; checked: boolean}>;
+
+    if (wizardData.workouts.length) {
+      unselectedItems = fitnessCategoriesWorkout.filter(
+        category =>
+          wizardData.workouts.findIndex(
+            workout => workout.label === category.label,
+          ) === -1,
+      );
+    } else {
+      unselectedItems = [...fitnessWorkouts];
+    }
+
+    setFitnessCategoriesWorkout([...wizardData.workouts, ...unselectedItems]);
+  }, []);
+
+  const onBlurFitnessInput = (value: string) => {
+    if (
+      value &&
+      !fitnessCategoriesWorkout.filter(
+        workout =>
+          workout.label.toLocaleLowerCase() === value.toLocaleLowerCase(),
+      ).length
+    ) {
+      setFitnessCategoriesWorkout([
+        {label: value, checked: true},
+        ...fitnessCategoriesWorkout,
+      ]);
+
+      dispatch(
+        updateUserData({
+          workouts: [...wizardData.workouts, {label: value, checked: true}],
+        }),
+      );
+    }
+  };
+
+  const onPressWorkoutItem = (workout: {label: string; checked: boolean}) => {
```

Comment:
The type `{label: string, checked: boolean` is seen 3 different times in this file, this is a red flag for duplication, please use `type` and `interface` avoid hardcoding it, it looks ugly and if you need to add or remove something to it, you'd have to cover all of the instances.